src/write_gcm.c

```diff
@@ -4158,6 +4158,27 @@ static int wg_write(const data_set_t *ds, const value_list_t *vl,
     }
     --queue->size;
     to_remove->next = NULL;
+    char metadata[8192];
+    metadata[0] = '{';
+    char *meta_ptr = metadata + 1;
+    int meta_size = sizeof(metadata) - 1;
```

Comment:
Should '1' be '2'? You're writing '{' and '}' for sure, plus the man pages for (s)snprintf say that it will write up to 'n-1' characters if you pass in size 'n'. sizeof(metadata) - 2 would match the actual amount of space available for writing.